invoices/sql_store.go

```diff
@@ -132,6 +132,14 @@ type SQLInvoiceQueries interface { //nolint:interfacebloat
 
 	GetInvoicePaymentHashByAddIndex(ctx context.Context,
 		addIndex sql.NullInt64) ([]byte, error)
+
+	ClearInvoiceHashIndex(ctx context.Context) error
+
+	GetMigration(ctx context.Context, migrationID string) (
+		sqlc.MigrationTracker, error)
+
+	UpdateMigration(ctx context.Context,
+		arg sqlc.UpdateMigrationParams) error
```

Comment:
adding on to my other comment: if we separate out migration code, then we dont need to add these to the main interface

Reply:
specifically things like `ClearInvoiceHashIndex`

Reply:
Since we still need the methods in `SQLInvoiceQueries` for insertion do you think it is worth adding the extra lines of code (extra interface and change of signatures for migration) for separation? I think we might just keep them here and remove in the next version we we also remove the temp table.

Reply:
> I think we might just keep them here and remove in the next version we we also remove the temp table.

I think im maybe struggling to picture this move - can you maybe just explain a bit more what we will do in the next version?